scripts/patch_release/patch.py

```diff
@@ -51,31 +69,147 @@ def store_file(
         syn (synapseclient.Synapse): The Synapse client object.
         new_path (str): The path to the file to be stored.
         new_release_synid (str): The Synapse ID of the release folder where the file will be stored.
-        release_name (str): The name of the release.
 
     Returns:
         None
     """
-    ent_name = os.path.basename(new_path.replace(f"_{release_name}", ""))
-    new_ent = synapseclient.File(new_path, name=ent_name, parentId=new_release_synid)
-    syn.store(new_ent)
+    new_ent = synapseclient.File(new_path, parentId=new_release_synid)
+    new_ent = syn.store(new_ent)
+    return new_ent
+
+
+def patch_file(syn: synapseclient.Synapse, synid: str, tempdir: str, new_release_synid: str, keep_values: pd.Series, column: str) -> str:
+    """
+    Patches a file in Synapse by filtering out rows based on the provided keep values.
+
+    Args:
+        syn (synapseclient.Synapse): The Synapse client object.
+        synid (str): The Synapse ID of the entity to be patched.
+        tempdir (str): The temporary directory to store the patched file.
+        new_release_synid (str): The Synapse ID of the release folder where the patched file will be stored.
+        keep_values (pd.Series): The values to keep in the dataframe.
+        column (str): The column name to filter on.
+    """
+    entity = syn.get(synid, followLink=True)
+    df = _filter_tsv(filepath=entity.path, keep_values=keep_values, column=column)
+    # Specific filtering fro the data gene matrix file because the string NA must
+    # replace the blank values
+    if entity.name == "data_gene_matrix.txt":
+        df[df.isnull()] = "NA"
+    # df = pd.read_csv(entity.path, sep="\t", comment="#")
+    # df = df[df[column].isin(keep_values)]
+    dftext = process_functions.removePandasDfFloat(df)
+    new_path = os.path.join(tempdir, os.path.basename(entity.path))
+    with open(new_path, "w") as o_file:
+        o_file.write(dftext)
+    store_file(syn, new_path, new_release_synid)
+    # TODO: return a named tuple so its not just returning the path
```

Comment:
What data would you return if not just the path in this case?

Reply:
Im going to update the comment due to YAGNI